### PR TITLE
feat: make proxy routing optional & update health check

### PR DESCRIPTION
**Pull request by oderayi**

### CODE REVIEW

File: src/domain/parties/getPartiesByTypeAndID.js [hunk 1]
```diff
@@ -97,7 +98,7 @@ const getPartiesByTypeAndID = async (headers, params, method, query, span, cache
     if (destination) {
       const destParticipantModel = await participant.validateParticipant(destination)
       if (!destParticipantModel) {
-        const proxyId = await proxyCache.lookupProxyByDfspId(destination)
+        const proxyId = proxyEnabled && await proxyCache.lookupProxyByDfspId(destination)
```
Comment (geka-evk) [inline src/domain/parties/getPartiesByTypeAndID.js:101]: Didn't we discuss to implement the logic for disabling proxyCache inside proxyCache?
 So if `proxyCacheConfig.enabled !== false`, any methods will return `null` or `false`

Comment (oderayi) [reply src/domain/parties/getPartiesByTypeAndID.js:101]: More like saying passing a flag into a database client to make it optional. That is not the responsibility of the client library.

File: src/handlers/health.js [hunk 2]
```diff
@@ -44,7 +57,8 @@ module.exports = {
    * responses: 200, 400, 401, 403, 404, 405, 406, 501, 503
    */
   get: async (context, request, h) => {
-    const health = await healthCheck.getHealth()
+    healthCheck.serviceChecks = getSubServices(Config, request)
+    const health = await healthCheck.getHealth({ request })
```
Comment (geka-evk) [inline src/handlers/health.js:61]: Can we avoid passing the whole request-object, and pass only needed data:
```
...
  const { isAdmin, proxyCache } = request.server.app
  healthCheck.serviceChecks = getSubServices(Config, isAdmin)
  const health = await healthCheck.getHealth(proxyCache)
  ...
```

File: src/lib/healthCheck/subServiceHealth.js [hunk 3]
```diff
@@ -56,6 +56,31 @@ const getSubServiceHealthDatastore = async () => {
...
 
+/**
+ * @function getProxyCacheHealth
+ *
+ * @description
+ *   Gets the health of the proxy cache by checking the connection to the cache.
+ *
+ * @returns Promise<SubServiceHealth> The SubService health object for the proxy cache
+ */
+const getProxyCacheHealth = async ({ request }) => {
```
Comment (geka-evk) [inline src/lib/healthCheck/subServiceHealth.js:67]: Can we avoid passing Hapi-specific _request_ object to this function, and pass only proxyCache?

File: src/domain/parties/getPartiesByTypeAndID.js (hunk 1 above)
Comment (oderayi) [reply src/domain/parties/getPartiesByTypeAndID.js:101]: Having the proxy cache in the execution path when its disabled and using its return values to decide a behavior we already know ahead of time seems worse to me.

**Review by geka-evk — APPROVED** [review]

**Pull request merged**